### PR TITLE
Add support for RoundHalfEven aka "Banker's Rounding"

### PR DESCRIPTION
**Pull request by floodfx**
Fixes: https://github.com/bojanz/currency/issues/19

### CODE REVIEW

File: amount.go [hunk 1]
```diff
@@ -26,6 +26,10 @@ const (
 	RoundUp
 	// RoundDown rounds towards 0, truncating extra digits.
 	RoundDown
+	// RoundHalfEven rounds up if the digits are > 0.5. If the digits are equal
```
Comment (bojanz) [inline amount.go:29]: Let's rephrase this as:
```
// RoundHalfEven rounds up if the next digit is > 5. If the next digit is equal
// to 5, it rounds to the nearest even decimal. Also called bankers' rounding.
```

File: amount_test.go [hunk 2]
```diff
@@ -529,6 +529,14 @@ func TestAmount_RoundTo(t *testing.T) {
...
 
+		{"12.344", 2, currency.RoundHalfEven, "12.34"},
+		{"12.345", 2, currency.RoundHalfEven, "12.34"},
+		{"12.346", 2, currency.RoundHalfEven, "12.35"},
+
+		{"12.334", 2, currency.RoundHalfEven, "12.33"},
+		{"12.335", 2, currency.RoundHalfEven, "12.34"},
+		{"12.336", 2, currency.RoundHalfEven, "12.34"},
+
 		// Negative amounts.
```
Comment (bojanz) [inline amount_test.go:540]: Can we also add test cases for negative amounts?

**Review by bojanz — CHANGES_REQUESTED** [review]
Thank you for your contribution! Added a few small comments, I hope you approve.
